lib/ecl/ecl_util.c

```diff
@@ -121,37 +121,49 @@ char * ecl_util_alloc_base_guess(const char * path) {
 }
 
 
+/*
+  The eclipse filenames must have either ALL_CAPITAL.EGRID or alternatviely
+  all_lowercase.egrid; mixed case is illegal. This function scans all the
+  characters after the last '/' and checks if they are all lowercase, or all
+  UPPER CASE.
+*/
```

Comment:
Why is mixed case illegal?

Reply:
Ask Schlumberger, I guess it is to ensure compliance on case insensitive filsystems

Reply:
Okay, but do _we_ have to follow Schlum here?  Can't we read a file called `JohanSverdrup.EGRID` without issues?

Reply:
That I agree - I think[1] we can read a file with any name if it is passed directly to the open function like:
```
grid = EclGrid("my_really_FUNNY_GRID.txt")
file = EclFile("MixedCase.UnRst")
```
However the functions this PR is about is related to create filename from components, infer types and so on - and then I think it makes sense to follow the conventions; but maybe not?


[1]: Will verify with a test!